builds/fabric8-che/plugins/keycloak-plugin-server/src/main/java/com/redhat/che/keycloak/server/KeycloakAuthenticationFilter.java

```diff
@@ -42,9 +44,13 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
                 || isInternalRequest(authHeader)) {
             LOG.debug("Skipping {}", requestURI);
             chain.doFilter(req, res);
-        } else {
+        } else if (KeycloakUserChecker.matchesUsername(authHeader)) {
             super.doFilter(req, res, chain);
             LOG.debug("{} status : {}", request.getRequestURL(), ((HttpServletResponse) res).getStatus());
+        } else {
```

Comment:
OK this is personal taste: I would prefer

    if(....){
         //dosomething1
         return;
    }
    if(....){
         //dosomething2
         return;
    }
    //dosomething3

than

    if(....){
         //dosomething1
    }
    else if(....){
         //dosomething2
    } else {
        //dosomething3
    }